virtme/commands/run.py

```diff
@@ -968,6 +969,46 @@ def cleanup_console_script():
     qemuargs.extend(["-device", "vhost-vsock-pci,guest-cid=%d" % args.port])
 
 
+def ssh_client(args):
+    user = f"{args.user}@" if args.user else ""
+
+    if args.remote_cmd is not None:
+        exec_escaped = args.remote_cmd.replace('"', '\\"')
+        remote_cmd = f'bash -c "{exec_escaped}"'
+    else:
+        remote_cmd = ""
+
+    cmd = ["ssh", "-p", str(args.port), f"{user}localhost"] + (
+        [remote_cmd] if remote_cmd else []
+    )
+    if args.dry_run:
+        print(" ".join(cmd))
+    else:
+        os.execvp("ssh", cmd)
+
+
+def ssh_server(args, arch, qemuargs, kernelargs):
+    # Check if we need to generate the ssh host keys for the guest.
+    ssh_key_dir = f"{CACHE_DIR}/.ssh"
+    if not os.path.exists(ssh_key_dir):
+        os.mkdir(ssh_key_dir)
+        os.system(f"ssh-keygen -t rsa -b 2048 -f {ssh_key_dir}/id_rsa -N '' -q")
+
+    # Implicitly enable dhcp to automatically get an IP on the network
+    # interface and prevent interface renaming.
+    kernelargs.extend(["virtme.dhcp", "net.ifnames=0", "biosdevname=0"])
+
+    # Tell virtme-ng-init / virtme-init to start sshd and use the current
+    # username keys/credentials.
+    username = get_username()
```

Comment:
What if `--user` is passed? Should it eventually be:

    username = args.user if args.user else get_username()

Or maybe you do want the current user, because if `--user root` is used, but `vng` is not launched as root, that will be an issue?

Reply:
I think ssh should be allowed only for the user that runs vng, independently on the `--user` option. Otherwise it's a bit too tricky to figure out file permissions, etc. and we may have more problems than benefits.... In fact I think I may also ignore `--user` in `--ssh-client` and just allow the current user to log into the vng instance for now. It seems safer and simpler and we can improve the logic later if needed. :)

Reply:
Fine for me to keep it simple for the moment. Should we have this "limitation" written somewhere?